File: bindings/swift/examples/GrpcServiceInfo.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct GrpcServiceInfoExample {
  static func main() async throws {
    let client = try GrpcClient.newTestnet()

    let info = try await client.serviceInfo()
    if let chainId = info.chainId {
      print("Chain ID:", chainId)
    }
    if let epoch = info.epoch {
      print("Epoch:", epoch)
    }
    if let checkpointHeight = info.executedCheckpointHeight {
      print("Checkpoint height:", checkpointHeight)
    }

    let gasPrice = try await client.referenceGasPrice()
    print("Reference gas price:", gasPrice)
  }
}

File: bindings/python/examples/grpc_service_info.py
# Copyright (c) 2026 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk import *

import asyncio


async def main():
    client = GrpcClient.new_localnet()

    info = await client.service_info()
    print("Chain ID:", info.chain_id)
    print("Epoch:", info.epoch)
    print("Checkpoint height:", info.executed_checkpoint_height)

    gas_price = await client.reference_gas_price()
    print("Reference gas price:", gas_price)


if __name__ == "__main__":
    asyncio.run(main())
